metrics/lib/common.bash

```diff
@@ -42,3 +42,54 @@ function save_results(){
 	bash $LIB_DIR/send_results.sh -n "$1" -a "$2" -r "$3" -u "$4"
 }
 
+# Find a reasonable path to the hypervisor on this system
+function get_qemu_path(){
+    ## Regular expressions used for TOML parsing
+    # Matches a section header
+    section_re="^\s*\[(\S+)]"
+    # Matches the name of the hypervisor section
+    hypervisor_re="^hypervisor\..*"
+    # Matches the variable containing the qemu path
+    qemu_re="^\s*path\s*=\s*\"([^\"]+)"
+
+    # Create a list of potential configuration files
+    declare -a conf_files
+    conf_files+=("${LIB_DIR}/../../../runtime/config/configuration.toml")
+    conf_files+=($(find /etc -type f -name configuration.toml -exec \
+        grep -l 'hypervisor.qemu' {} + 2>/dev/null))
+
+    # Check the potential files sequentially
+    for ((i=0; i<${#conf_files[@]}; i++)); do
+        conf_file="${conf_files[$i]}"
+        if [[ -f "${conf_file}" ]]; then
+            # Attempt to parse the found config file (TOML)
+            while read line; do
+                if [[ $line =~ $section_re ]]; then
+                    # New section
+                    section=${BASH_REMATCH[1]}
+                elif [[ $section =~ $hypervisor_re ]]; then
+                    # Look for qemu path
+                    if [[ $line =~ $qemu_re ]]; then
+                        # Found it
+                        qemu_path="${BASH_REMATCH[1]}"
+
+                        # Check it
+                        if [[ -x "$qemu_path" ]]; then
+                            # This looks like a valid qemu; stop looking
+                            break 2 # Get out of both loops
+                        fi
+                    fi
+                fi
+            done < $conf_file
+        fi
+    done
+
+    # Check whether we got a good result
+    if [[ "" == "$qemu_path" ]]; then
+        die Failed to find qemu path in $conf_file
+    elif [[ ! -x "$qemu_path" ]]; then
```

Comment:
apart from we may have different machine types between different runtimes - and ideally this code would 'just work' with other modified or experimental VMs. I'm happy to skip this check here, and let that fail in the runtime itself - and/or, if it does not fail in the runtime then the runtime is doing something we don't understand ;-)
It is a little swings-and-roundabouts - I wish we didn't have to rely on such internal knowledge of how the runtime works in the first place, but to do some of the metrics measures (footprint of all the components, cpu consumption of similar) we do.